### DISCUSSION TITLE
Disable the `no-implicit-coercion` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#220] [General]
There is a proposal to disable the `no-implicit-coercion` rule.

> `no-implicit-coercion`: introduces unnecessary verbosity
> \- @edgardz

**Comment by leroykorterink** [comment]
There are a lot of different ways to convert value types. Some of them might be hard to read and understand. Such as:

```ts
const b = !!foo;
const b = ~foo.indexOf(".");
const n = +foo;
const n = 1 * foo;
const s = "" + foo;
foo += ``;
```

Those can be replaced with the following code:

```ts
const b = Boolean(foo);
const b = foo.indexOf(".") !== -1;
const n = Number(foo);
const n = Number(foo);
const s = String(foo);
foo = String(foo);
```

Please see [the documentation](https://eslint.org/docs/latest/rules/no-implicit-coercion) of the rule.

**Comment by ThaNarie** [comment]
> introduces unnecessary verbosity

In this case, it comes with better clarity; being explicit about what's happening, what's intended, without ambiguity (or chance for errors).

Some examples where I like it being used (over `!!`):

```tsx
<div isActive={Boolean(someValue && callSomeFunction(someValue))} />

<div>
 {Boolean(child) && cloneElement(child as ReactElement, { ref: setBounds })}
</div>

const isMultiline = !isInline && Boolean(leftOne ?? rightOne);

<RowHeader ref={ref} onClick={onClick} $isClickable={Boolean(onRowClick)} {...props}>
```
